src/responsivePropBuilder.ts

```diff
@@ -1,8 +1,17 @@
+import { Threshold } from './ThresholdMap';
 import defaultThresholdMap from './defaultThresholdMap';
 
-const getCurrentValue = value => (value !== undefined ? value : null);
+const getCurrentValue = (value: any) => (value !== undefined ? value : null);
 
-const responsivePropBuilder = (currentThreshold, props, configuration, thresholdMap = defaultThresholdMap) => {
+export interface GenericProps {
+  [key: string]: any;
+}
+
+export interface ResponsivePropsConfig {
+  propKeys: Array<string>;
```

Comment:
`Array<T>` is used a couple times in this PR, but it's generally better to use `T[]`.